### PR TITLE
docs: partition_manager: Clarify one_of with span values

### PR DESCRIPTION
**Pull request by kyberdin**
State explicitly in the documentation that spans cannot be used for the values of one_of dicts.
See the associated Jira issue for more information.

Ref: NCSDK-4620
Signed-off-by: Stephen Stauts <stephen.stauts@nordicsemi.no>

### CODE REVIEW

**Review by hakonfam — APPROVED** [review]

File: scripts/partition_manager/partition_manager.rst [hunk 1]
```diff
@@ -327,6 +327,7 @@ RAM partition configuration
           size: 0xa000
 
 All occurrences of a partition name can be replaced with a dict with the key ``one_of``, which is resolved to the first existing partition in the ``one_of`` value.
+The value of the ``one_of`` key must be a list of placeholder or image partitions, and it cannot be a span.
```
Comment (tejlmand) [inline scripts/partition_manager/partition_manager.rst:330]: in addition to this, would it be possible to detect a malformed pm.yml here:
https://github.com/nrfconnect/sdk-nrf/blob/master/scripts/partition_manager.py

and fail gracefully with a proper error message to the user ?

Comment (kyberdin) [reply scripts/partition_manager/partition_manager.rst:330]: Sure, I think that would be a very useful feature to add.

I think adding it here is out of scope of this Jira task and PR, but I would be happy to open a separate Jira task for implementing this functionality down the road.

Comment (kyberdin) [reply scripts/partition_manager/partition_manager.rst:330]: Opened NCSDK-7398 to address this feature implementation.

**Review by tejlmand — APPROVED** [review]

**Review by FrancescoSer — COMMENTED** [review]
Waiting for this https://github.com/nrfconnect/sdk-nrf/pull/3420 to be merged before continuing with the docreview

**Comment by kyberdin** [comment]
@FrancescoSer Branch has been rebased on top of most recent master, which includes the merged doc changes from #3420 .

**Review by FrancescoSer — APPROVED** [review]

**Comment by kyberdin** [comment]
@FrancescoSer Your changes should now be correctly squashed into the original commit.

**Pull request merged**